src/wasm2js.h

```diff
@@ -2048,13 +2050,13 @@ Ref Wasm2JSBuilder::processFunctionBody(Module* m,
 
 void Wasm2JSBuilder::addMemoryFuncs(Ref ast, Module* wasm) {
   Ref memorySizeFunc = ValueBuilder::makeFunction(WASM_MEMORY_SIZE);
-  memorySizeFunc[3]->push_back(ValueBuilder::makeReturn(
-    makeAsmCoercion(ValueBuilder::makeBinary(
-                      ValueBuilder::makeDot(ValueBuilder::makeName(BUFFER),
-                                            IString("byteLength")),
-                      DIV,
-                      ValueBuilder::makeInt(Memory::kPageSize)),
-                    AsmType::ASM_INT)));
+  memorySizeFunc[3]->push_back(ValueBuilder::makeReturn(makeAsmCoercion(
+    ValueBuilder::makeBinary(
+      ValueBuilder::makeDot(ValueBuilder::makeName(BUFFER),
+                            IString("byteLength")),
+      RSHIFT,
+      ValueBuilder::makeInt(31 - CountLeadingZeroes(Memory::kPageSize))),
```

Comment:
can't this just be `Log2(memory::kPageSize)`?

Reply:
need extend `Log2`. It currently hardcoded only up to `32` argument. But yeah, it's probably best function for that

Reply:
I agree, improving Log2 seems like the best plan.